src/apps/vpn/controller.cpp

```diff
@@ -576,7 +578,15 @@ void Controller::handshakeTimeout() {
 
 void Controller::disconnected() {
   logger.debug() << "Disconnected from state:" << m_state;
+  mozilla::glean::session::session_end.set();
 
+  // TODO: This must be after submission of ping - ensure it comes after the submissino
```

Comment:
Are you going to address this Todo in this PR?

Reply:
It's a note for https://mozilla-hub.atlassian.net/browse/VPN-4443. I updated the wording to remove the TODO, and make it clear it's a note for when working that ticket.